test/models/vrp_rejection_test.rb

```diff
@@ -0,0 +1,159 @@
+# Copyright © Mapotempo, 2021
+#
+# This file is part of Mapotempo.
+#
+# Mapotempo is free software. You can redistribute it and/or
+# modify since you respect the terms of the GNU Affero General
+# Public License as published by the Free Software Foundation,
+# either version 3 of the License, or (at your option) any later version.
+#
+# Mapotempo is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+# or FITNESS FOR A PARTICULAR PURPOSE.  See the Licenses for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with Mapotempo. If not, see:
+# <http://www.gnu.org/licenses/agpl.html>
+#
+require './test/test_helper'
+
+module Models
+  class VrpRejectionTest < Minitest::Test
```

Comment:
I am not sure splitting VRP tests into `VrpRejectionTest` and `VrpTest` is a good idea. I believe they belong to one place but I might be wrong. I don't know what the best practice is but I suspect tests belonging the same class usually stay together.

But if we want to go down that road, I think we should move at least tests like `test_reject_if_shipments_and_periodic_heuristic` and `test_reject_if_rests_and_periodic_heuristic` and possibly others to here as well because the logic is exactly the same.

Reply:
As method on model is called `check_consistency` I suggest you to call this file `vrp_consistency_test.rb`. But as said by Halil, you had to merge `scheduling_instance_validity_test.rb ` in this file.

Reply:
This makes sense but at the same time it is contradictory to what we discussed : dividing files in order to have smaller tests files. I think this is a big subject to discuss, to make things properly and with one defined guide line.

As @senhalil suggested I think this PR should only move non api tests from test/api/v01/vrp_tests.rb.

Reply:
I don't think it's a BIG subject. Please don't waste time on it. You can split multiple files in this one PR.
In addition (but this can be performed in another PR), `check_consistency` method should be split in a concern (separated file) from the model.

Reply:
I tried to demonstrate in my analysis that we have bigger fishes to catch. That's why I suggested that we fix the error of non-api tests in this PR quickly (since it is an error) then decide what to do with other files (in a general fashion) because this is clearly not an issue with one single test file or one single class. It is a prevalent problem in our codebase. Dividing test files hastily will not resolve the main problem and it will not help us in the long run. That's why I suggest a separate proper rework PR.

I think our main problem is that our models are "empty", we can't have useful tests "for" them. For example models/vehicles.rb, models/services.rb or models/units.rb, they don't have "their" validation logic in them as member functions. So when we need to test "detect_unfeasible_service" or "check_consistency" we are forced to create a VRP and either post it or call wrapper_vrp on it. THIS is the source of the issue which leads to big test files.